File: support/mock_backend/wiringPiD/network.c
#include <stdint.h>
#include <stddef.h>
#include <stdbool.h>
#include <string.h>
#include "wiringPiD/network.h"
char * getClientIP(void) { return "127.0.0.1"; }
int setupServer(int serverPort) { (void)serverPort;return 1; }
int sendGreeting(int clientFd) { (void)clientFd;return 1; }
int sendChallenge(int clientFd) { (void)clientFd;return 1; }
int getResponse(int clientFd) { return clientFd+100; }
int passwordMatch(const char * password) { (void)password;return 1; }
void closeServer(int clientFd) { (void)clientFd; }

File: tests/wiringPiD/network.py
import importlib
import pytest

m=importlib.import_module("wiringop.wiring_pi_d.network")

def test_get_client_ip():
    _=m.get_client_ip()

def test_get_response_legacy():
    assert m.get_response_legacy(3)==103

def test_setup_server():
    _=m.setup_server(1)

def test_send_greeting():
    _=m.send_greeting(1)

def test_send_challenge():
    _=m.send_challenge(1)

def test_get_response():
    _=m.get_response(1)

def test_password_match():
    _=m.password_match("x")

def test_close_server():
    _=m.close_server(1)
